Mathlib/LinearAlgebra/RootSystem/Defs.lean

```diff
@@ -380,29 +380,57 @@ lemma pairing_reflection_perm_self_right (i j : ι) :
     sub_add_cancel_left, ← toLin_toPerfectPairing, map_neg, toLin_toPerfectPairing,
     root_coroot_eq_pairing]
 
+/-- If `R` is an `S`-algebra, a root pairing over `R` is said to be valued in `S` if the pairing
+between a root and coroot always belongs to `S`.
+
+Of particular interest is the case `S = ℤ`. See `RootPairing.IsCrystallographic`. -/
+class IsValuedIn (S : Type*) [CommRing S] [Algebra S R] : Prop where
```

Comment:
```suggestion
class IsValuedIn (S : Type*) [CommSemiRing S] [Algebra S R] : Prop where
```
This will allow `S = Nat`, which will be nice when stating/proving
```
P.coxeterWeightIn Nat i j ∈ ({0, 1, 2, 3, 4} : Set Nat)
```

Reply:
Agreed; this might also be useful in a potential (though not currently planned) refactor of `RootPairing.Base` (which I now I think is "correct" only for crystallographic pairings).

Reply:
On second thought, I don't think this will ever be useful: the pairing cannot avoid negative values (though the Coxeter weights can).

Reply:
Aah of course, silly me :facepalm: